### DISCUSSION TITLE
Issue getting collection sales data

### DISCUSSION THREAD

**Discussion by farairo** [ProjectOpenSea/opensea-js#1227] [Q&A]
I am trying to fetch collections sales data using this endpoint v2/events/collection/{collection_slug} but the data returned is missing lot of sales for some reason. For example I tried to fetch sales data for collection "Bored Ape Yacht Club" here https://docs.opensea.io/reference/list_events_by_collection but its missing lot of recent sales.

I cant understand what i am doing wrong here, any help would be really appreciated

**Comment by ryanio** [comment]
can you share some specific examples of missing events to help us debug? are there sales that are showing on OpenSea website but not returning through the endpoint?

**Reply by farairo** [reply]
yes for example i tried to fetch sales for "Bored Ape Yacht Club" collection and its missing recent 40 to 50 sales that i can see on opensea website.
the latest sales I get using the endpoint is 18 hours old

**Reply by thenerdassassin** [reply]
@farairo Are you using the `next` cursor to paginate through the results?

Take the `next` field returned in the first call and pass it in as a query parameter to the second call.

**Reply by farairo** [reply]
@thenerdassassin yes i am using next value to paginate through the results.

should not the sales data be sorted by recent sales ? the first sales is 18 hours old

**Comment by farairo** [comment]
Could it be possible the sales I am missing did not happen on opensea marketplace? although i see them on opensea site

**Comment by farairo** [comment]
@ryanio @thenerdassassin I can confirm now that the missing collection sales are the once that has happend on other marketplaces.
 does that mean opensea api does not show sales from other marketplaces even though we can see the sales on opensea site?
